test/blackbox/common/DDSBlackboxTestsBasic.cpp

```diff
@@ -271,6 +279,191 @@ TEST(DDSBasic, MultithreadedReaderCreationDoesNotDeadlock)
+                            msg->buffer[pid_pos + 1] = 0xff;
+                        }
+                        break;
+                    }
+                    case (PID_RELATED_SAMPLE_IDENTITY):
+                    {
+                        if (plength >= 24)
+                        {
+                            ParameterSampleIdentity_t p(pid, plength);
+                            if (!fastdds::dds::ParameterSerializer<ParameterSampleIdentity_t>::read_from_cdr_message(p,
+                                    msg, plength))
+                            {
+                                return false;
+                            }
+                            exists_pid_related_sample_identity = true;
+                        }
+                        break;
+                    }
+
+                    default:
+                        break;
+                }
+                return true;
+            };
+
+    uint32_t original_pos = msg.pos;
+    bool is_sentinel = false;
+    while (!is_sentinel)
+    {
+        msg.pos = original_pos + qos_size;
+
+        ParameterId_t pid{PID_SENTINEL};
+        uint16_t plength = 0;
+        bool valid = true;
+        auto msg_pid_pos = msg.pos;
+        valid &= fastrtps::rtps::CDRMessage::readUInt16(&msg, (uint16_t*)&pid);
+        valid &= fastrtps::rtps::CDRMessage::readUInt16(&msg, &plength);
+
+        if (pid == PID_SENTINEL)
+        {
+            // PID_SENTINEL is always considered of length 0
+            plength = 0;
+            is_sentinel = true;
+        }
+
+        qos_size += (4 + plength);
+
+        // Align to 4 byte boundary and prepare for next iteration
+        qos_size = (qos_size + 3) & ~3;
+
+        if (!valid || ((msg.pos + plength) > msg.length))
+        {
+            return false;
+        }
+        else if (!is_sentinel)
+        {
+            if (!parameter_process(&msg, pid, plength, msg_pid_pos))
+            {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+/**
+ * This test checks that PID_RELATED_SAMPLE_IDENTITY and
+ * PID_CUSTOM_RELATED_SAMPLE_IDENTITY are being sent as parameter,
+ * and that the new PID_RELATED_SAMPLE_IDENTITY is being properly
+ * interpreted.
+ * Inside the transport filter, both PIDs are indentified, and the old PID is overwritten.
+ * Reader only receives the new PID, and identifies the related sample identity.
+ */
+TEST(DDSBasic, PidRelatedSampleIdentity)
+{
+    PubSubWriter<HelloWorldPubSubType> reliable_writer(TEST_TOPIC_NAME);
+    PubSubReader<HelloWorldPubSubType> reliable_reader(TEST_TOPIC_NAME);
+
+    // Test transport will be used in order to filter inlineQoS
+    auto test_transport = std::make_shared<eprosima::fastrtps::rtps::test_UDPv4TransportDescriptor>();
+    bool exists_pid_related_sample_identity = false;
+    bool exists_pid_custom_related_sample_identity = false;
+    bool another_debug_bool_flag = false;
```

Comment:
This does not seem to be used for anything other than passing it to the lambda which then does not use it.